samtranslator/model/stepfunctions/events.py

```diff
@@ -161,11 +184,19 @@ def _construct_target(self, resource, role, dead_letter_queue_arn=None):  # type
             if self.Target and "Id" in self.Target
             else generate_valid_target_id(self.logical_id, EVENT_RULE_SFN_TARGET_SUFFIX)
         )
-        target = {
-            "Arn": resource.get_runtime_attr("arn"),
-            "Id": target_id,
-            "RoleArn": role.get_runtime_attr("arn"),
-        }
+
+        if provided_role:
+            target = {
+                "Arn": resource.get_runtime_attr("arn"),
+                "Id": target_id,
+                "RoleArn": provided_role,
+            }
+        elif role:
+            target = {
+                "Arn": resource.get_runtime_attr("arn"),
+                "Id": target_id,
+                "RoleArn": role.get_runtime_attr("arn"),
+            }
```

Comment:
```suggestion
        target = {
            "Arn": resource.get_runtime_attr("arn"),
            "Id": target_id,
            "RoleArn": provided_role if provided_role else role.get_runtime_attr("arn"),
        }
```
Think something like this looks cleaner